packages/cli-tool/src/commands/generate/index.ts

```diff
@@ -90,24 +93,20 @@ export default class Generate extends Command {
     appName: string,
     uiFramework: string,
   ): Promise<void> => {
-    if (uiFramework === 'bootstrap') {
-      cli.info('Configure Bootstrap...')
-      const result = await this.config.runHook('install-bootstrap', {
```

Comment:
I think @carryall [is right](https://github.com/nimblehq/react-templates/pull/102#discussion_r915429018), and better to split these ifs to separate functions.

Reply:
Continuing both discussions in this thread.

IMO what makes this function more complex that it should be is the Hook system. I had a quick discussion with @hoangmirs — we both think our context does not benefits from the advantages of using a hook (child process, hot-reloadable, etc...) but we get the cons (code more complex to read and typescripts error received late).

Replacing the hook with exported functions we could do:
- 1 folder `ui-framework` with an index.ts file that contain `setupUIFramework(uiFramework)` method.
- 1 file per framework (similar to the hooks files)

What do you think? Would that be a good solution? Any alternative? 🙂

cc @carryall @Nihisil

Reply:
I made a "try" in this branch (the PR was created only to display the code changes) :) 
https://github.com/nimblehq/react-templates/pull/108

Reply:
@malparty That looks great to me, and I also think maybe we should move the `setUIFramework` function to the `ui-framework/index.ts` too as same as I moved the `setVersionControl` in [the PR](https://github.com/nimblehq/react-templates/pull/95) 💭  (I still nee to adjust a bit the folder and file name 😅)

Reply:
Oh yeah, I missed this part. 
Ok, I will review the file structure accordingly and ping here once both PR are up & running :)

Reply:
Ok, rebased AND refactored according to the "replace hook with functions" idea! 🚀